### PR TITLE
fixed #10863 Update tableNameOrAlias to equalsIgnoreCase

### LINKED ISSUE #10863: Inconsistent case check problem
### Which version of ShardingSphere did you use?
4.1.1 & 5.0.0-RC1-SNAPSHOT

There is an inconsistency in case handling in class `TablesContext` and `ProjectionsContextEngine`, which needs to be corrected.

image

### PR DESCRIPTION
**Pull request by zoujianchao**
Fixes #10863

Update tableNameOrAlias to equalsIgnoreCase

### CODE REVIEW

**Comment by tristaZero** [comment]
@RaigorJiang needs your attention here.

File: shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/table/TablesContextTest.java [hunk 1]
```diff
@@ -90,7 +90,16 @@ public void assertFindTableNameWhenColumnSegmentOwnerAbsentAndSchemaMetaDataCont
         assertTrue(actual.isPresent());
         assertThat(actual.get(), is("table_1"));
     }
-    
+
```
Comment (RaigorJiang) [inline shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/table/TablesContextTest.java:93]: Please Keep indents (including blank lines) consistent with the previous one.

File: shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/table/TablesContextTest.java [hunk 2]
```diff
@@ -90,7 +90,16 @@ public void assertFindTableNameWhenColumnSegmentOwnerAbsentAndSchemaMetaDataCont
...
+    @Test
+    public void assertFindTableNameWhenTableNameOrAliasIgnoreCase() {
+        SimpleTableSegment tableSegment1 = createTableSegment("table_1", "tbl_1");
+        SimpleTableSegment tableSegment2 = createTableSegment("table_2", "tbl_2");
+        Optional<String> actual = new TablesContext(Arrays.asList(tableSegment1, tableSegment2)).findTableNameFromSQL("Tbl_1");
+        assertTrue(actual.isPresent());
+        assertThat(actual.get(), is("table_1"));
+    }
+
```
Comment (RaigorJiang) [inline shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/table/TablesContextTest.java:102]: indent problem too

File: shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/select/projection/engine/ProjectionsContextEngineTest.java [hunk 3]
```diff
@@ -391,4 +391,49 @@ private void assertCreateProjectionsContextWithTemporaryTable(final SelectStatem
                 .createProjectionsContext(selectStatementContext.getFromSimpleTableSegments(), projectionsSegment, groupByContext, new OrderByContext(Collections.emptyList(), false));
         assertNotNull(actual);
     }
+
+    @Test
+    public void assertCreateProjectionsContextWhenTableNameOrAliasIgnoreCaseForMySQL() {
+        assertCreateProjectionsContextWhenTableNameOrAliasIgnoreCase(new MySQLSelectStatement(), new MySQLSelectStatement());
+    }
+
```
Comment (RaigorJiang) [inline shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/select/projection/engine/ProjectionsContextEngineTest.java:399]: These indents are not consistent with the previous one.

**Comment by RaigorJiang** [comment]
Hi @zoujianchao 
It's a good work, just a few indents problems, please fix them, thanks

**Comment by zoujianchao** [comment]
ok, I'll do it

File: shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/select/projection/engine/ProjectionsContextEngineTest.java (hunk 3 above)
Comment (zoujianchao) [reply shardingsphere-infra/shardingsphere-infra-binder/src/test/java/org/apache/shardingsphere/infra/binder/segment/select/projection/engine/ProjectionsContextEngineTest.java]: my idea, it's not flexible

**Review by RaigorJiang — APPROVED** [review]
It is a good first PR now, believe you must have gained something, congratulations!

**Comment by zoujianchao** [comment]
Thank you. I'll try my best

**Review by RaigorJiang — APPROVED** [review]

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**